### ISSUE TITLE
GraphQL support should be a third party application.

### ISSUE THREAD

**Issue by lovelydinosaur** [Kludex/starlette#619]
I'd like our graphql support to move out into being a third party package.

We should then document:

* The third party package it's been moved to.
* Links to ASGI implementations with ariadne, tartiflette, strawberry.

Questions:

* Which of Ariadne, Tartiflette & Strawberry currently have provide an ASGI application, that Starlette can mount?
* Are there any other GraphQL implementations I've missed here?
* Anyone keen to take on the GraphQL implementation as a third party package?

**Comment by lovelydinosaur** [comment]
* Ariadne ASGI app: https://ariadnegraphql.org/docs/asgi
* Tatiflette: https://github.com/tartiflette/tartiflette-starlette

**Comment by dhirschfeld** [comment]
`strawberry` asgi app:
* https://github.com/strawberry-graphql/strawberry/blob/master/docs/05_integrations/asgi.md

**Comment by euri10** [comment]
> `strawberry` asgi app:
> * https://github.com/strawberry-graphql/strawberry/blob/master/docs/05_integrations/asgi.md

`​@​strawberry​.​field` forever : 💗 !

**Comment by JayH5** [comment]
Just commenting to add some updated links to libraries that can provide GraphQL support to Starlette applications. These will be in the Starlette documentation as well:
* [Ariadne](https://ariadnegraphql.org/docs/asgi)
* [`tartiflette-asgi`](https://tartiflette.github.io/tartiflette-asgi/)
* [Strawberry](https://strawberry.rocks/docs/integrations/asgi)
* [`starlette-graphene3`](https://github.com/ciscorn/starlette-graphene3)

**Comment by RoyTrudell** [comment]
> > `strawberry` asgi app:
> > 
> > * https://github.com/strawberry-graphql/strawberry/blob/master/docs/05_integrations/asgi.md
> 
> `​@​strawberry​.​field` forever : 💗 !

Dead Link

**Comment by patrick91** [comment]
> > > `strawberry` asgi app:
> > > 
> > > * https://github.com/strawberry-graphql/strawberry/blob/master/docs/05_integrations/asgi.md
> > 
> > 
> > `​@​strawberry​.​field` forever : 💗 !
> 
> Dead Link

Updated link is here: https://strawberry.rocks/docs/integrations/asgi ☺️

**Comment by cglacet** [comment]
How easy would it be to replace the existing solution with any of the suggested alternatives? It looks like the current solution is quite strongly coupled to graphene, or am I wrong? 

The idea would be to output the schema as plain text or json to a third party library?

I would gladly see even over-simplified example to appear in the documentation because its not clear to me how everything is interacting. For example I can see in strawberry's case that:

> 
> The GraphQL app accepts two options at the moment:
> 
> * schema: mandatory, the schema created by `strawberry.Schema`.

Seems like a problem since most people using starlette probably used graphene to generate their schema. I'm personally "stuck" with graphene as I used graphene-sqlalchemy, but maybe I missed a better option to interface my SQLAlchemy models with GraphQL.

I found an example in Ariadne's documentation: [Starlette integration](https://ariadnegraphql.org/docs/0.4.0/starlette-integration) where they use the string version of the schema (aka `type_defs`). There is a discussion about this [here](https://github.com/mirumee/ariadne/issues/115).
